CMakeLists.txt

```diff
@@ -8,6 +8,9 @@ find_package(Asio REQUIRED)
 find_package(fmt REQUIRED)
 find_package(spdlog REQUIRED)
 
+set(CMAKE_ARCHIVE_OUTPUT_DIRECTORY lib)
```

Comment:
I swear these weren't here the first review. :P
Please move to a separate commit, also move binaries to bin for consistency, and wrap in an if checking that this is the top-level project. I've more than once been annoyed that some random library I pull in via FetchContent randomly changed my output dirs.